ksz8863/src/ksz8863_ctrl.c

```diff
@@ -57,56 +59,48 @@ static inline bool bus_unlock(void)
 static esp_err_t ksz8863_i2c_write(uint8_t reg_addr, uint8_t *data, size_t len)
 {
     esp_err_t ret = ESP_OK;
-    i2c_port_t i2c_port = s_ksz8863_ctrl_intf->i2c_bus_spec.i2c_port;
-    uint8_t dev_addr = s_ksz8863_ctrl_intf->i2c_bus_spec.dev_addr;
-
-    i2c_cmd_handle_t cmd = i2c_cmd_link_create();
-    ESP_RETURN_ON_FALSE(cmd, ESP_ERR_NO_MEM, TAG, "I2C link create error");
-    ESP_GOTO_ON_ERROR(i2c_master_start(cmd), err, TAG, "I2C master start error");
-    ESP_GOTO_ON_ERROR(i2c_master_write_byte(cmd, dev_addr | I2C_MASTER_WRITE, true), err, TAG, "I2C master write error");
-    ESP_GOTO_ON_ERROR(i2c_master_write_byte(cmd, reg_addr, true), err, TAG, "I2C master write error");
-    if (likely(reg_addr != KSZ8863_RESET_ADDR)) {
-        ESP_GOTO_ON_ERROR(i2c_master_write(cmd, data, len, true), err, TAG, "I2C master write error");
-    } else {
-        // when SW reset is performed, KSZ does not generate ACK
-        ESP_GOTO_ON_ERROR(i2c_master_write(cmd, data, len, false), err, TAG, "I2C master write error");
-    }
-    ESP_GOTO_ON_ERROR(i2c_master_stop(cmd), err, TAG, "I2C master stop error");
-    // Lock since multiple MAC/PHY instances exist and the KSZ may be also accessed by user (MAC tables, etc...)
-    ESP_GOTO_ON_FALSE(bus_lock(KSZ8863_I2C_LOCK_TIMEOUT_MS), ESP_ERR_TIMEOUT, err, TAG, "I2C bus lock timeout");
-    ESP_GOTO_ON_ERROR(i2c_master_cmd_begin(i2c_port, cmd, pdMS_TO_TICKS(KSZ8863_I2C_TIMEOUT_MS)), err_release, TAG,
-                      "I2C master command begin error");
+    uint8_t *reg_addr_and_data = malloc(len + 1);
+
+    i2c_device_config_t dev_cfg = {
+        .dev_addr_length = I2C_ADDR_BIT_LEN_7,
+        .device_address = s_ksz8863_ctrl_intf->i2c_bus_spec.dev_addr >> 1,
+                .scl_speed_hz = s_ksz8863_ctrl_intf->i2c_bus_spec.scl_speed_hz,
+                .flags.disable_ack_check = (reg_addr == KSZ8863_RESET_ADDR)
```

Comment:
Please keep the comment as in original code.

Reply:
Unable to. `i2c_master` doesn't provide the ability to turn off ACK check after creating the device, so now it is disabled for all operations, not only for reset.

Reply:
I'm not sure if it is safe approach. I noticed you tried to find workaround. Does it work? Other option could be creating 2 I2C devices, one with ACK check disable for reset. Have you considered it? Anyway, I would suggest go over documentation if this situation is addressed and if not, try to contact author of the driver.

Reply:
Considered, but I imagine having `i2c_handle`, `i2c_noackcheck_handle` and `spi_handle` would be awful to read in the future. Especially if `i2c_noackcheck_handle` will only be used in one line of code. I guess contacting the driver's author might be the best choice here.